### ISSUE TITLE
[1.19.2] Collar enchantment duplication glitch

### ISSUE THREAD

**Issue by B4CKU** [AlexModGuy/DomesticationInnovation#144]
Mod version: 1.6.1
Minecraft version: 1.19.2
Forge version: 43.2.3

Some collared pets, will keep their collars and their enchantments, when swapped out, while still dropping said collars on the ground. 

I'm not sure, but this could be related to the issue #104 

Video: 

https://github.com/AlexModGuy/DomesticationInnovation/assets/50628162/70c2e8f4-3dc6-4358-aa10-1ba603e817ac

List of all used mods:

![image](https://github.com/AlexModGuy/DomesticationInnovation/assets/50628162/f8fa06c9-a8de-4f92-860f-0ff7596b458d)

**Comment by crafting-Aaron** [comment]
Yeah it is still a problem. The only workaround is adding those two enchantments at the end of a maxed collar on a NEW pet.

**Comment by SiverDX** [comment]
The fix in #104 only fixes race conditions which are server-side

It always allows updates from the client-side and since `onLivingUpdate` runs for both sides the client-side will call the `sync` methods for those (and other enchants) while bypassing the block logic

Not syncing those two from the client-side withing `onLivingUpdatE` fixes the issue
It also doesn't seem to break the rendering parts of those enchants (at least for the blazing enchant, I don't know what the other one does)

**Comment by mc-goodch** [comment]
This is still an issue in the 1.20 version (domesticationinnovation-1.7.0-1.20). The herding enchantment cannot be removed and is causing chaos. Adding a blank collar tag pops the enchanted tag off of the mob, but the enchantments remain. Adding another blank to the mob pops off a fully enchanted tag. The only way to end this is to kill the mob.
